### PR TITLE
Add type definitions for Timer 

### LINKED ISSUE #427: Add TimerTrigger types
Specifically, add a type to be used for `myTimer` instead of `any`:

```typescript
const timerTrigger: AzureFunction = async function (context: Context, myTimer: any): Promise<void> {
```

Need to keep in mind that starting in v2 of the worker (v3 of the host), TimerTrigger [changed from](https://docs.microsoft.com/en-us/azure/azure-functions/functions-versions?tabs=javascript%2Cv4#breaking-changes-between-2x-and-3x) camelCase to PascalCase in v2 of the worker (v3 of the host)

### PR DESCRIPTION
**Pull request by jamiehaywood**
relates #398 

I wasn't entirely sure about the definitions, but I had a look at the source code, which does have comments, and lifted and shifted from there:
https://github.com/Azure/azure-webjobs-sdk-extensions/blob/9c87af74f90159f8b987c77aa77546f55a62a84e/src/WebJobs.Extensions/Extensions/Timers/Scheduling/ScheduleStatus.cs

### CODE REVIEW

**Comment by PCOffline** [comment]
Are there any updates regarding this PR, or is the Azure team simply ignoring it?

**Comment by anthonychu** [comment]
@alrod @AnatoliB could you please take a look?

**Comment by jamiehaywood** [comment]
bump @anthonychu @alrod @AnatoliB

**Comment by PCOffline** [comment]
bump @anthonychu @alrod @AnatoliB

**Comment by jamiehaywood** [comment]
I'm gonna keep bumping this until someone from Microsoft replies. I appreciate you're all busy, but: 
1. This isn't a complex PR.
2. I've followed the contribution guidelines.
3. It will genuinely improve DX with this library.

cc: @Sam-Rowe @fortunkam - do you know anyone that can move this along?

File: types/public/Interfaces.d.ts
Comment (ejizba) [inline types/public/Interfaces.d.ts]: These should be strings, not Dates

File: types/public/Interfaces.d.ts
Comment (ejizba) [inline types/public/Interfaces.d.ts:248]: `scheduleStatus` and `isPastDue` should be at the same level as `schedule`, rather than inside it

Screen Shot 2021-10-04 at 3 30 17 PM

File: src/public/Interfaces.ts
Comment (ejizba) [inline src/public/Interfaces.ts:273]: We actually do not use the "master" branch. I just opened a PR to update the docs on the readme to more accurately describe the branching strategy: https://github.com/Azure/azure-functions-nodejs-worker/pull/424. Please choose either the "v1.x" or "v2.x" branch as the target for your PR. 

Personally, I would recommend the "v2.x" branch. The properties for a timer trigger changed from PascalCase to camelCase starting with v2 of the worker (v3 of the host) and it appears you are using camelCase. That change is documented [here](https://docs.microsoft.com/azure/azure-functions/functions-versions?tabs=javascript%2Cv4#breaking-changes-between-2x-and-3x).

**Review by ejizba — CHANGES_REQUESTED** [review]
Hi @jamiehaywood thanks for contributing! I just joined the team and will be focusing on the Node.js experience, including a lot of improvements to the TypeScript types. We apologize for the delay on this PR and general lack of updates on the types, but the tide is about to turn for the better!

File: types/public/Interfaces.d.ts
Comment (PCOffline) [reply types/public/Interfaces.d.ts]: In what format?

File: src/public/Interfaces.ts
Comment (PCOffline) [reply src/public/Interfaces.ts:273]: Regarding the versioning, especially v2 and v3, the docs are **very** unclear on the distinguish between the versions. Half of the time there is no docs for v3 at all, despite v3 not being backwards compatible with v2 in many cases. Is there a reason for that?

Comment (ejizba) [reply src/public/Interfaces.ts:273]: This is the main doc regarding versions: 
https://docs.microsoft.com/azure/azure-functions/functions-versions

With this section on breaking changes:
https://docs.microsoft.com/azure/azure-functions/functions-versions?tabs=javascript%2Cv4#breaking-changes-between-2x-and-3x

Do you have examples or suggestions of where you expect to see more docs on versioning? Would be happy to get those updated.

File: types/public/Interfaces.d.ts
Comment (ejizba) [reply types/public/Interfaces.d.ts]: Here's what I'm seeing for the format:
Screen Shot 2021-10-04 at 4 05 41 PM

**Comment by jamiehaywood** [comment]
@ejizba thanks for the comments, I will resolve as soon as I can. thank you again.

**Comment by jamiehaywood** [comment]
hi @PCOffline & @ejizba - applied the changes as requested. Please let me know if there are any other changes that need to be made before merge and release.

**Comment by PCOffline** [comment]
Seems good to me, but I'm not a maintainer so perhaps ejizba will have something to add

**Review by ejizba — APPROVED** [review]
Looks good - thank you so much!

**Comment by ejizba** [comment]
Hey @jamiehaywood and @PCOffline thanks again for getting this in! I've released an update to the npm package. See https://github.com/Azure/azure-functions-nodejs-worker/issues/398#issuecomment-961507389 for more info

**Comment by PCOffline** [comment]
Glad to be a part of this! Thanks for the surprisingly helpful advice :)

**Pull request merged**
